docs/user/managing-devices.md

```diff
@@ -0,0 +1,586 @@
+
+| Parameter | Description |
+| --------- | ----------- |
+| Name | A user-defined name for the application. This will be used when the web UI and CLI list applications. |
+| Image | A reference to an application package in an OCI registry. |
+| EnvVars | (Optional) A list of key/value-pairs that will be passed to the deployment tool as environment variables or command line flags. |
+
+For each application in the "applications" section of the device's specification, there exist a corresponding device status information that contains the following information:
+
+| Status Field | Description |
+| ------------ | ----------- |
+| | |
+
+### Managing Applications on the Web UI
+
+### Managing Applications on the CLI
+
+To deploy an application package from an OCI registry, specify it in the device's `spec.applications[]` as follows:
+
+```yaml
+apiVersion: v1alpha1
+kind: Device
+metadata:
+  name: some_device_name
+spec:
+[...]
+  applications:
+  - name: wordpress
+    image: quay.io/flightctl-demos/wordpress-app:latest
+    envVars:
+      WORDPRESS_DB_HOST: "mysql"
+      WORDPRESS_DB_USER: "user"
+      WORDPRESS_DB_PASSWORD: "password"
+[...]
+```
+
+To deploy an unpackaged application from a Git repository, specify it in the device's `spec.applications[]` as follows:
+
+```yaml
+apiVersion: v1alpha1
+kind: Device
+metadata:
+  name: some_device_name
+spec:
+[...]
+  applications:
+  - name: wordpress
+    git:
+      url: https://github.com/flightctl/flightctl-demos.git
+      revision: main
+      path: /wordpress
+    envVars:
+      WORDPRESS_DB_HOST: "mysql"
+      WORDPRESS_DB_USER: "user"
+      WORDPRESS_DB_PASSWORD: "password"
+[...]
+```
+
+To deploy an unpackaged application inline with the device specification, specify it in the device's `spec.applications[]` as follows:
+
+```yaml
+apiVersion: v1alpha1
+kind: Device
+metadata:
+  name: some_device_name
+spec:
+[...]
+  applications:
+  - name: wordpress
+    inline:
+      podman-compose.yaml: |
+        version: “3.7"
+        services:
+          wordpress:
+            image: “wordpress:latest”
+        [...]
+    envVars:
+      WORDPRESS_DB_HOST: "mysql"
+      WORDPRESS_DB_USER: "user"
+      WORDPRESS_DB_PASSWORD: "password"
+[...]
+```
+
+## Using Device Lifecycle Hooks
+
+You can specify device lifecycle hooks, which are custom actions the agent shall perform before or after certain lifecycle events. For example, you can add a shell script to your OS images that backs up your application data and then specify that this script shall be run and completed successfully before the agent can start updating the system.
+
+The following lifecycle hooks are supported:
+
+| Lifecycle Hook | Description |
+| -------------- | ----------- |
+| beforeUpdating | Specified actions are executed just before the agent starts updating the system. |
+| afterUpdating | Specified actions are executed after the agent has completed updating the system. |
```

Comment:
Today implementation does not align with the docs. beforeUpdating is tracked as `OnBeforeUpdate` and is fired before a file is updated by the config controller vs before updating the system. `OnAfterUpdate` refers to immediately after a file is updated.

Reply:
Both the original PR #424 and the "fixup" PR #449 call the hook `beforeUpdating` and give an example of `run: /usr/local/bin/create-backup.sh` that only makes sense if the hook gets called before updating the system. So I'm not clear what actually is / isn't implemented.

My suggestion would be to remove this whole section until we've clarified the API.

Reply:
Removed.